soc/xtensa/intel_adsp/common/include/adsp-clk.h

```diff
@@ -35,8 +35,13 @@ int cavs_clock_set_freq(uint32_t freq_idx);
 struct cavs_clock_info *cavs_clocks_get(void);
 
 /* Device tree defined constants */
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
+#define ADSP_DfPMCCU          DT_REG_ADDR(DT_NODELABEL(dfpmccap))
+#define ADSP_SHIM_CLKCTL        (*((volatile uint32_t *)(ADSP_DfPMCCU + 0x78)))
```

Comment:
Please no hardcoded addresses for SHIM_CLKCTL.
Consider adding new devicetree node nested inside the dfpmccap node with address of 0x78
(nested nodes have starting address derived from parent node).
something like this:
```
dfpmccap: dfpmccap@71b00 {
    compatible = "intel,adsp-dfpmccap";
    reg = <0x71b00 0x4000000>;

    adsp_shim_clkctl: ace_shim_clkctl@78 { };
};
```
Or alternatively, you can try using phandles and reference clkctl address inside shim_clkctl node in reg property, something like this I think:
```
clkctl: clkctl {
    compatible = "intel,adsp-shim-clkctl";
    adsp-clkctl-clk-wovcro = <0>;
    adsp-clkctl-clk-lpro = <1>;
    adsp-clkctl-clk-hpro = <2>;
    adsp-clkctl-freq-enc = <0xc 0x0 0x4>;
    adsp-clkctl-freq-mask = <0x0 0x0 0x0>;
    adsp-clkctl-freq-default = <2>;
    adsp-clkctl-freq-lowest = <0>;
    wovcro-supported;
};

//note - no node address after @
adsp_shim_clkctl: ace_shim_clkctl {
    reg = <(&dfpmccap + 0x78)>
}
```
This according to example from [Devicetree - writing property values](https://docs.zephyrproject.org/3.1.0/build/dts/intro.html#writing-property-values) should:
* take address of dfpmccap and add 0x78 offset to it like described in mathematical operators example.

Reply:
This is fine, we have tons of registers and it would result in super big dtsi if you want to describe every register there.
Currently in dtsi we have blocks of registers like reg = <0x0007c000 0x1000> for gpdma
or shim: shim@71f00 reg = <0x71f00 0x100>;
which is similar to DfPMCCU

Reply:
ace_dfpmccu struct was added and used dfclkctl as offset 0x78:
https://github.com/zephyrproject-rtos/zephyr/pull/52205/files#diff-4b3ead25ea95f174fec287a37688bd60e5ee557ece7405d6be0862dd0483b52b